### PR TITLE
k3d-action implementation

### PR DESCRIPTION
**Pull request by kuritka**
Documentation will come during the next PR. 
`.github/workflows` contains several use cases supported by action:

 - single cluster
 - single cluster with default docker registry
 - two clusters within one subnet
 - two clusters within one subnet with default docker registry
 - two clusters within two subnets
 - two clusters within two subnets with default docker registry
 - two pairs of clusters within two subnets
 - two pairs of clusters within two subnets with default docker registry

Action also supports custom registry configuration, by setting `k3d-registry-config-path `; [see this](https://rancher.com/docs/k3s/latest/en/installation/private-registry/#mirrors).

All is configurable via several attributes (please look into `.github/workflows` for demo examples)

  k3d-name:
  k3d-args:  [k3d command tree](https://k3d.io/usage/commands/)
  k3d-network: network name used in scenarios where you need multiple subnets, otherwise action provides default subnet
  k3d-subnet:  subnet CIDR, can be reused when previous step created network
  k3d-registry:  Injects default k3d private registry. No TLS, no credentials.
  k3d-registry-config-path: Path to config file where you can specify [custom configuration](https://rancher.com/docs/k3s/latest/en/installation/private-registry/#with-tls) for private registry

Ping me if needed

### CODE REVIEW

File: run.sh [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+usage(){
+  cat <<EOF
+
+  Usage: $(basename "$0") <COMMAND>
+  Commands:
+      deploy            deploy custom k3d cluster
+
+  Environment variables:
+      deploy
+                        K3D_NAME (Required) k3d cluster name.
```
Comment (somaritane) [inline run.sh]: `K3D_NAME` looks a bit ambiguous, consider using `K3D_CLUSTER` or `K3D_CLUSTER_NAME` instead

File: action.yaml [hunk 2]
```diff
@@ -0,0 +1,47 @@
+name: Setup K3d
+description: K3d action with multi-cluster support.
+author: AbsaOSS
+# see: https://haya14busa.github.io/github-action-brandings/
+branding:
+  icon: grid
+  color: red
+inputs:
+  k3d-name:
```
Comment (somaritane) [inline action.yaml]: Consider using:
```suggestion
  k3d-cluster-name:
```

**Review by somaritane — COMMENTED** [review]
Looks awesome @kuritka!
I've just added few comments about naming.

**Comment by kuritka** [comment]
@somaritane thx, fixed

File: run.sh [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+#######################
+if [[ "$#" -lt 1 ]]; then
+  usage
+  exit 1
+fi
+if [[ -z "${NO_COLOR}" ]]; then
+      YELLOW="\033[0;33m"
+      CYAN="\033[1;36m"
+      NC="\033[0m"
+      RED="\033[0;91m"
```
Comment (ytsarev) [inline run.sh:162]: nitpich from shellcheck
```
shellcheck run.sh

In run.sh line 156:
      RED="\033[0;91m"
      ^-^ SC2034: RED appears unused. Verify use (or export if used externally).

```

File: action.yaml [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    description: "Detected k3d network subnet"
+    value: ${{ steps.main.outputs.k3d-subnet }}
+runs:
+  using: composite
+  steps:
+    - id: main
+      shell: bash
+      run: ${{ github.action_path }}/run.sh deploy
+      env:
+        K3D_CLUSTER_NAME: ${{ inputs.k3d-cluster-name }}
```
Comment (ytsarev) [inline action.yaml]: well, technically it is k3s cluster but I wouldn't go down that road :D

Comment (kuritka) [reply action.yaml]: @ytsarev , true. I'll rename

Comment (ytsarev) [reply action.yaml]: @kuritka totally up to you, but yeah it is k3s

**Review by k0da — APPROVED** [review]
LATM

**Review by somaritane — APPROVED** [review]
lgtm!

File: action.yaml (hunk 4 above)
Comment (kuritka) [reply action.yaml]: @somaritane, @ytsarev  What do you think about complete renaming? env vars will refactor as well

k3d-cluster-name -> cluster-name
k3d-args -> args
k3d-network -> network
k3d-subnet -> subnet-CIDR
k3d-registry -> use-default-registry
k3d-registry-config-path -> override-registry-config-path

Comment (ytsarev) [reply action.yaml]: @kuritka looks good to me!

**Pull request merged**